exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/Foreman.java

```diff
@@ -115,6 +115,8 @@
   private static final ObjectMapper MAPPER = new ObjectMapper();
   private static final long RPC_WAIT_IN_MSECS_PER_FRAGMENT = 5000;
 
+  private Thread myThreadRef; // Thread that is currently executing the Foreman.
```

Comment:
Is it really necessary ? myThreadRef is used to force the foreman thread to throw an exception when calling moveToState. Any other thread will ignore that (no matter what value myThreadRef has) and block on the latch instead.